surelock/src/main/java/com/smashingboxes/surelock/Surelock.java

```diff
@@ -228,11 +237,15 @@ public void enrollFingerprintAndStore(String key, byte[] valueToEncrypt,
     public void enrollFingerprintAndStore(String key,
                                           SurelockFragment surelockFragment,
                                           @NonNull FragmentManager fragmentManager,
-                                          String fingerprintDialogFragmentTag) {
+                                          String fingerprintDialogFragmentTag) throws SurelockException {
         initKeyStoreKey();
         Cipher cipher;
         try {
-            cipher = initCipher(Cipher.ENCRYPT_MODE);
+            try {
+                cipher = initCipher(Cipher.ENCRYPT_MODE);
+            } catch (InvalidKeyException | UnrecoverableKeyException | KeyStoreException e) {
+                throw new SurelockException("Failed to init Cipher for encryption", e);
```

Comment:
Yep, that's exactly it. At this point, if the key is invalid, it's invalid due to an unusual issue, since the key is created just above this on line 241.
If the key is invalid here in any way, then there are way bigger issues. But, the likelihood of this happening is probably very low (at this point in the flow).